spynnaker/pyNN/models/neuron/synapse_io.py

```diff
@@ -387,47 +439,54 @@ def _parse_static_data(row_data, dynamics):
-            undelayed_connections = dynamics.read_static_synaptic_data(
-                post_vertex_slice, n_synapse_types, ff_size, ff_data)
-            undelayed_connections["source"] += pre_vertex_slice.lo_atom
-            connections.append(undelayed_connections)
-
-        if delayed_row_data is not None and delayed_row_data.size:
-            ff_size, ff_data = self._parse_static_data(
-                delayed_row_data, dynamics)
-            delayed_connections = dynamics.read_static_synaptic_data(
-                post_vertex_slice, n_synapse_types, ff_size, ff_data)
-
-            # Use the row index to work out the actual delay and source
-            n_synapses = dynamics.get_n_synapses_in_rows(ff_size)
-            synapse_ids = range(len(n_synapses))
-            row_stage = numpy.array([
-                i // pre_vertex_slice.n_atoms
-                for i in synapse_ids], dtype="uint32")
-            row_min_delay = (row_stage + 1) * 16
-            connection_min_delay = numpy.concatenate([
-                numpy.repeat(row_min_delay[i], n_synapses[i])
-                for i in synapse_ids])
-            connection_source_extra = numpy.concatenate([
-                numpy.repeat(
-                    row_stage[i] * numpy.uint32(pre_vertex_slice.n_atoms),
-                    n_synapses[i])
-                for i in synapse_ids])
-            delayed_connections["source"] -= connection_source_extra
-            delayed_connections["source"] += pre_vertex_slice.lo_atom
-            delayed_connections["delay"] += connection_min_delay
-            connections.append(delayed_connections)
+        ff_size, ff_data = SynapseIORowBased._parse_static_data(
+            row_data, dynamics)
+        undelayed_connections = dynamics.read_static_synaptic_data(
+            post_vertex_slice, n_synapse_types, ff_size, ff_data)
+        undelayed_connections["source"] += pre_vertex_slice.lo_atom
+        return undelayed_connections
 
-        return connections
+    @staticmethod
+    def _read_delayed_static_data(
+            dynamics, pre_vertex_slice, post_vertex_slice,
+            n_synapse_types, delayed_row_data):
+        if delayed_row_data is None or not delayed_row_data.size:
+            return numpy.zeros(
+                0, dtype=AbstractSynapseDynamics.NUMPY_CONNECTORS_DTYPE)
+
+        ff_size, ff_data = SynapseIORowBased._parse_static_data(
+            delayed_row_data, dynamics)
+        delayed_connections = dynamics.read_static_synaptic_data(
+            post_vertex_slice, n_synapse_types, ff_size, ff_data)
+
+        # Use the row index to work out the actual delay and source
+        n_synapses = dynamics.get_n_synapses_in_rows(ff_size)
+        synapse_ids = range(len(n_synapses))
+        row_stage = numpy.array([
+            i // pre_vertex_slice.n_atoms
+            for i in synapse_ids], dtype="uint32")
+        row_min_delay = (row_stage + 1) * 16
```

Comment:
cloned from line 477 to 489 with lines 544 and 557

Reply:
Looks like this has gone